numpy/core/tests/test_nep50_promotions.py

```diff
@@ -61,12 +64,116 @@ def test_nep50_examples():
     assert res.dtype == np.float64
 
 
-@pytest.mark.xfail
+@pytest.mark.parametrize("dtype", np.typecodes["AllInteger"])
+def test_nep50_weak_integers(dtype):
+    # Avoids warning (different code path for scalars)
+    np._set_promotion_state("weak")
```

Comment:
Should the tests reset global state at finish?

Reply:
Yes, there is a module wide auto-fixture that defaults to `"weak_and_warn"` and resets the state at the end.